R/scrape_statcast.R

```diff
@@ -3,72 +3,84 @@
+#' @param playerid The MLBAM ID for the player whose data you want to query.
+#' @param player_type The player type. Can be 'batter' or 'pitcher'. Default is 'batter'
+#' @param ... currently ignored
 #' @keywords MLB, sabermetrics, Statcast
 #' @importFrom utils read.csv
+#' @importFrom tibble tribble
+#' @importFrom dplyr mutate_ filter_
+#' @importFrom lubridate year
 #' @export
 #' @examples
 #' \dontrun{
-#' scrape_statcast_savant(start_date = "2016-04-06", 
-#'   end_date = "2016-04-15", playerid = 621043, player_type='batter')
+#' correa <- scrape_statcast_savant(start_date = "2016-04-06", 
+#'   end_date = "2016-04-15", playerid = 621043)
 #'
-#' scrape_statcast_savant(start_date = "2016-04-06", 
+#' noah <- scrape_statcast_savant(start_date = "2016-04-06", 
 #'   end_date = "2016-04-15", playerid = 592789, player_type='pitcher')
 #'
-#' scrape_statcast_savant(start_date = "2016-04-06", end_date = "2016-04-06")
+#' daily <- scrape_statcast_savant(start_date = "2016-04-06", end_date = "2016-04-06")
 #' }
 
-scrape_statcast_savant <- function(start_date, end_date, playerid=NULL, player_type=NULL) {
-  # Check to make sure args are in the correct format.
-  if(!is.character(start_date) | !is.character(end_date)) {
-    warning("Please wrap your dates in quotations in 'yyyy-mm-dd' format.")
-    return(NULL)
-  }
+scrape_statcast_savant <- function(start_date = Sys.Date() - 1, end_date = Sys.Date(), 
+                                   playerid = NULL, 
+                                   player_type = "batter", ...) UseMethod("scrape_statcast_savant")
+
+#' @rdname scrape_statcast_savant
+#' @export
+
+scrape_statcast_savant.Date <- function(start_date = Sys.Date() - 1, end_date = Sys.Date(), 
+                                        playerid = NULL, player_type = "batter", ...) {
   # Check for other user errors.
-  if(as.Date(start_date)<="2015-03-01") { # March 1, 2015 was the first date of Spring Training.
+  if (start_date <= "2015-03-01") { # March 1, 2015 was the first date of Spring Training.
     message("Some metrics such as Exit Velocity and Batted Ball Events have only been compiled since 2015.")
   }
-  if(as.Date(start_date)<="2008-03-25") { # March 25, 2008 was the first date of Spring Training.
+  if (start_date <= "2008-03-25") { # March 25, 2008 was the first date of Spring Training.
     stop("The data are limited to the 2008 MLB season and after.")
     return(NULL)
   }
-  if(as.Date(start_date)==Sys.Date()) {
+  if (start_date == Sys.Date()) {
     message("The data are collected daily at 3 a.m. Some of today's games may not be included.")
   }
-  if(as.Date(start_date)>as.Date(end_date)) {
+  if (start_date > as.Date(end_date)) {
     stop("The start date is later than the end date.")
     return(NULL)
   }
+  
+  playerid_var <- ifelse(player_type == "pitcher", 
+                         "pitchers_lookup%5B%5D", "batters_lookup%5B%5D")
+  
+  vars <- tibble::tribble(
```

Comment:
Added to the tibble that is created. This allows for any games to be included, including post season and spring training. Some fields may be unnecessary, but added just in case. 

```
vars <- tibble::tribble(
  ~var, ~value, 
  "all", "true",
  "hfPT", "", 
  "hfAB", "", 
  "hfBBT", "", 
  "hfPR", "", 
  "hfZ", "", 
  "stadium", "", 
  "hfBBL", "", 
  "hfNewZones", "",
  "hfGT", "R%7CPO%7CS%7C&hfC",
  "hfSea", paste0(lubridate::year(start_date), "%7C"),
  "hfSit", "", 
  "hfOuts", "", 
  "opponent", "", 
  "pitcher_throws", "", 
  "batter_stands", "", 
  "hfSA", "", 
  "player_type", player_type,
  "game_date_gt", as.character(start_date),
  "game_date_lt", as.character(end_date),
  "hfInfield", "", 
  "team", "", 
  "position", "", 
  "hfOutfield", "", 
  "hfRO", "", 
  "home_road", "",
  "pitchers_lookup%5B%5D", ifelse(is.null(592789), "", 592789), 
  "hfFlag", "", 
  "hfPull", "", 
  "metric_1", "", 
  "hfInn", "", 
  "min_pitches", 0,
  "min_results", 0,
  "group_by", "name",
  "sort_col", "pitches",
  "player_event_sort", "h_launch_speed",
  "sort_order", "desc",
  "min_abs", 0,
  "type", "details"
) %>%
  dplyr::mutate_(pairs = ~paste(var, "=", value, sep = ""))
```

Reply:
Sure, but what is this about?
```
  "pitchers_lookup%5B%5D", ifelse(is.null(592789), "", 592789), 
```

Reply:
I guess ultimate it would be cool if users could pass values for all of these values through the `...`. Or maybe the function could take a `params` list?

Reply:
I literally kept telling myself "make sure to copy the right one, make sure to copy the right one". 

Re: all the values, yeah that was my thought. Figured it was worth putting them all in now.